### ISSUE TITLE
Unable to locate Javascript steps generated from building Typescript files

### ISSUE THREAD

**Issue by Billy-Stroud** [cucumber/language-service#187]
### 👓 What did you see?

I am using the [VSCode Cucumber plugin](https://marketplace.visualstudio.com/items?itemName=CucumberOpen.cucumber-official) within a Typescript project that imports common shared steps from a Typescript npm module that is packaged as Javascript.  

The plugin is able to successfully locate step definitions in Typescript from the local src code, but it fails to locate the shared Javascript steps from the dependency.  This seems to be due to the regular expression for matching Javascript step definitions not working for Javascript files that were transpiled down from Typescript.  Here's an example of what one of my steps looks like after it has been built into Javascript: 
```javascript
const cucumber_1 = require("@cucumber/cucumber");
(0, cucumber_1.Given)(/^I login$/, function () {
    return __awaiter(this, void 0, void 0, function* () {
        yield this.login();
    });
});
```

This seemed like it could be fixed by overriding TSX's `defineStepDefinitionQueries` [value](https://github.com/cucumber/language-service/blob/b1b0585de7879825e541b26c7ef3d12f71bcf156/src/language/tsxLanguage.ts#L83) in the [Javascript implementation](https://github.com/cucumber/language-service/blob/b1b0585de7879825e541b26c7ef3d12f71bcf156/src/language/javascriptLanguage.ts) to look for `Given|When|Then|Given)|When)|Then)` however I get `Invalid regular expression: Unmatched ')'` errors when testing with it even if I try to escape the parentheses.  I wasn't really sure what else to try in this case.  

Any help would be greatly appreciated!

### ✅ What did you expect to see?

Expected the plugin to find the Javascript steps.

### 📦 Which tool/library version are you using?

- Cucumber VSCode plugin: v1.8.1
- Node: v18.17.1
- Typescript: v5.3.3
- Cucumber: v9.6.0

### 🔬 How could we reproduce it?

1. Create a JS file with the following data: 
```javascript
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
const cucumber_1 = require("@cucumber/cucumber");
(0, cucumber_1.Given)(/^I login$/, function () {
    return __awaiter(this, void 0, void 0, function* () {
        yield this.login();
    });
});
```
2. Check the Cucumber Language Server output to see if it picks up the new step

### 📚 Any additional context?

----
 
*This text was originally generated from a [template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates), then edited by hand. [You can modify the template here.](https://github.com/cucumber/.github/edit/main/.github/ISSUE_TEMPLATE/bug_report.md)*

**Comment by barrymichaeldoyle** [comment]
I was having the same problem.

This PR seems to fix it for my projects:

https://github.com/cucumber/language-service/pull/186/files

**Comment by finnmerlett** [comment]
> I was having the same problem.
> 
> This PR seems to fix it for my projects:
> 
> https://github.com/cucumber/language-service/pull/186/files

This is unrelated to the original issue - it only deals with stripping out js/ts class decorators

I have done the actual fix in PR #222, adding a second javascript tree-sitter query to match compiled js step formats

**Comment by JBASTM** [comment]
Hi @finnmerlett, did you know if this fix is planned to be available in vscode extension soon ?
